CMakeLists.txt

```diff
@@ -53,6 +53,19 @@ if(LCM_ENABLE_JAVA)
 endif()
 
 # Python
+if(LCM_ENABLE_PYTHON AND APPLE)
+  find_program(PYTHON_CONFIG_EXECUTABLE python-config)
+  if (NOT PYTHON_CONFIG_EXECUTABLE)
+    message(SEND_ERROR "python-config executable not found, but python is required.")
```

Comment:
I don't think this is going to work... at least not reliably. If Python is enabled automatically, `LCM_ENABLE_PYTHON` won't be defined yet, but `lcm_option` will find Python and enable the bindings (using the wrong Python).

I think a better approach would be to create a `FindPython.cmake` that finds both — using this 'hint' logic for the libraries — and only sets `Python_FOUND` if both are found. This would also address the quirk that `lcm-python/CMakeLists.txt` needs to do a `find_package(PythonInterp REQUIRED)` which is somewhat inconsistent with only making the bindings available if the dependencies are found.

(As @bradking noted in RobotLocomotion/drake#2731, the optimal long-term fix is for CMake itself to get a better FindPython.)